task-ballerina/commons.bal

```diff
@@ -14,118 +14,54 @@
+// Represents a ballerina job, which can be used to execute periodically
+public type Job object {
 
-# Possible types of parameters that can be passed into the `AppointmentTaskPolicy`.
-public type AppointmentMisfirePolicy SMART_POLICY|DO_NOTHING|FIRE_AND_PROCEED|IGNORE_POLICY;
+  // Calls by the Scheduler when a trigger fires
+  public function execute();
+};
+
+# Policies related to a trigger.
+#
+# + errorPolicy - The policy, which is used to inform what it should do when a error occurs.
+# + waitingPolicy - The policy, which is used to inform what it should do when waiting to get a resource
+#                   for the trigger.
+public type TaskPolicy record {|
+   ErrorPolicy? errorPolicy = LOG_AND_TERMINATE;
+   WaitingPolicy? waitingPolicy = IGNORE;
+|};
 
-# Possible types of parameters that can be passed into the `RecurringTaskPolicy`.
-public type RecurringTaskPolicy SMART_POLICY|IGNORE_POLICY|FIRE_NEXT_WITH_EXISTING_COUNT|
-FIRE_NEXT_WITH_REMAINING_COUNT|FIRE_NOW_WITH_EXISTING_COUNT|FIRE_NOW_WITH_REMAINING_COUNT;
+# Possible types of parameters that can be passed into the `ErrorPolicy`.
+public enum ErrorPolicy {
+  LOG_AND_TERMINATE,
+  LOG_AND_CONTINUE,
+  TERMINATE,
+  CONTINUE
+}
 
-# Possible types of parameters that can be passed into the `OneTimeTaskPolicy`.
-public type OneTimeTaskPolicy SMART_POLICY|IGNORE_POLICY|FIRE_NOW;
+# Possible types of parameters that can be passed into the `WaitingPolicy`.
+public enum WaitingPolicy {
+  WAIT,
+  IGNORE,
+  LOG_AND_IGNORE
+}
 
-# The value of this constant is `smartPolicy`.
-public const string SMART_POLICY = "smartPolicy";
-# The value of this constant is `doNothing`.
-public const string DO_NOTHING = "doNothing";
-# The value of this constant is `fireAndProceed`.
-public const string FIRE_AND_PROCEED = "fireAndProceed";
-# The value of this constant is `fireNextWithExistingCount`.
-public const string FIRE_NEXT_WITH_EXISTING_COUNT = "fireNextWithExistingCount";
-# The value of this constant is `fireNextWithRemainingCount`.
-public const string FIRE_NEXT_WITH_REMAINING_COUNT = "fireNextWithRemainingCount";
-# The value of this constant is `fireNowWithExistingCount`.
-public const string FIRE_NOW_WITH_EXISTING_COUNT = "fireNowWithExistingCount";
-# The value of this constant is `fireNowWithRemainingCount`.
-public const string FIRE_NOW_WITH_REMAINING_COUNT = "fireNowWithRemainingCount";
-# The value of this constant is `fireNow`.
-public const string FIRE_NOW = "fireNow";
-# The value of this constant is `ignorePolicy`.
-public const string IGNORE_POLICY = "ignorePolicy";
+public isolated function getTimeInMillies(time:Time time) returns int|TaskError {
+    time:Time cTime = time:currentTime();
+    string timeZone = cTime.zone.id;
+    time:Time|error value = time:toTimeZone(time, timeZone);
+    if (value is time:Time) {
+        return value.time;
+    } else {
+        return error TaskError("Couldn't convert given time to milli seconds");
```

Comment:
Shall we append the error message to the new message. Otherwise, we lose the context. use String Template literal  to append

Reply:
This logic will change slightly with the new Time design

Reply:
Updated.